catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/catalog/CatalogApplication.java

```diff
@@ -372,7 +372,8 @@ private Response addDocument(
       URI uri = new URI(requestUrl.toString());
       UriBuilder uriBuilder = new UriBuilderImpl(uri).path("/" + id);
 
-      res.redirect(uriBuilder.build().toString(), HttpStatus.SC_CREATED);
+      res.status(HttpStatus.SC_CREATED);
+      res.header("Location", uriBuilder.build().toString());
       res.header(Metacard.ID, id);
```

Comment:
❓ Could you give a bit of background on the reason for this Location header being added? This was not added in the old RestEndpoint and isn't in the new one either, so why is this necessary now?

Reply:
It is added in the RESTEndpoint. https://github.com/codice/ddf/blob/3cc3d8c027cdde26936765490fb58cf969ed672f/catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java#L353

`res.redirect` added the location but for some reason the `res.header(Metacard.ID, id);` line under it wasn't getting called.

Reply:
Oops I missed that in the `RestEndpoint`. It could be that the `redirect` method call is meant to be a terminating call. Not sure.